src/simulator.zig

```diff
@@ -412,6 +410,8 @@ pub const Simulator = struct {
 
         simulator.cluster.state_checker.assert_cluster_convergence();
 
+        var checkpoint_id: ?u128 = null;
```

Comment:
This variable seems to be never assigned to

Reply:
Ahh I "refactored" at the last minute :facepalm:

Reply:
Hopefully the very next version of zig will catch these things!

Reply:
I reworked this -- the checkpoint id is now tested in its own loop _after_ the "repairing" loop. That means it can `assert` that the checkpoint ids match instead of just `return`ing if they don't.